File: Source/pushwoosh/android/src/main/java/com/pushwoosh/plugin/PushwooshNotificationServiceExtension.java
package com.pushwoosh.plugin;

import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;

import com.pushwoosh.internal.utils.JsonUtils;
import com.pushwoosh.internal.utils.PWLog;
import com.pushwoosh.notification.NotificationServiceExtension;
import com.pushwoosh.notification.PushMessage;

import org.json.JSONException;

public class PushwooshNotificationServiceExtension extends NotificationServiceExtension {
    private String TAG = "PushwooshNotificationServiceExtension";
    private boolean showForegroundPush = false;

    public PushwooshNotificationServiceExtension() {
        try {
            String packageName = getApplicationContext().getPackageName();
            ApplicationInfo ai =  getApplicationContext().getPackageManager().getApplicationInfo(packageName, PackageManager.GET_META_DATA);

            if (ai != null && ai.metaData != null) {
                showForegroundPush = ai.metaData.getBoolean("PW_BROADCAST_PUSH", false) || ai.metaData.getBoolean("com.pushwoosh.foreground_push", false);
            }
        } catch (PackageManager.NameNotFoundException e) {
            PWLog.error(TAG, "Failed to read AndroidManifest metaData", e);
        }

        PWLog.debug(TAG, "showForegroundPush = " + showForegroundPush);
    }

    @Override
    protected boolean onMessageReceived(PushMessage pushMessage) {
        boolean appOnForeground = isAppOnForeground();
        try {
            PushwooshPlugin.onMessageReceived(JsonUtils.jsonToMap(pushMessage.toJson()), !appOnForeground);
        } catch (JSONException e) {
            PWLog.error(TAG, "Failed to parse PushMessage from JSON", e);
        }
        return !PushwooshPlugin.showForegroundPush && appOnForeground || super.onMessageReceived(pushMessage);
    }

    @Override
    protected void onMessageOpened(PushMessage pushMessage) {
        boolean appOnForeground = isAppOnForeground();
        try {
            PushwooshPlugin.onMessageAccepted(JsonUtils.jsonToMap(pushMessage.toJson()), !appOnForeground);
        } catch (JSONException e) {
            PWLog.error(TAG, "Failed to parse PushMessage from JSON", e);
        }
    }
}

File: Source/pushwoosh/android/src/main/kotlin/com/pushwoosh/plugin/PushwooshNotificationServiceExtension.kt
package com.pushwoosh.plugin


import com.pushwoosh.internal.utils.PWLog
import android.content.pm.PackageManager
import com.pushwoosh.notification.NotificationServiceExtension
import com.pushwoosh.notification.PushMessage
import com.pushwoosh.internal.utils.JsonUtils


class PushwooshNotificationServiceExtension : NotificationServiceExtension() {
    private var showForegroundPush: Boolean = false
    private val TAG = "PushwooshNotificationServiceExtension"

    init {
        try {
            val packageName = applicationContext?.packageName
            val ai =  applicationContext?.packageManager?.getApplicationInfo(packageName, PackageManager.GET_META_DATA)

            if (ai?.metaData != null) {
                showForegroundPush = ai.metaData.getBoolean("PW_BROADCAST_PUSH", false) || ai.metaData.getBoolean("com.pushwoosh.foreground_push", false)
            }
        } catch (e: Exception) {
            PWLog.error(TAG, "Failed to read AndroidManifest metaData", e)
        }

        PWLog.debug(TAG, "showForegroundPush = $showForegroundPush")
    }


    override fun onMessageReceived(pushMessage: PushMessage): Boolean {
        val appOnForeground = isAppOnForeground()
        PushwooshPlugin.onMessageReceived(JsonUtils.jsonToMap(pushMessage.toJson()), !appOnForeground)
        return !PushwooshPlugin.showForegroundPush && appOnForeground || super.onMessageReceived(pushMessage)
    }


    override fun onMessageOpened(pushMessage: PushMessage) {
        val appOnForeground = isAppOnForeground()
        PushwooshPlugin.onMessageAccepted(JsonUtils.jsonToMap(pushMessage.toJson()), !appOnForeground)
    }
}